ethportal-api/src/types/consensus/beacon_state.rs

```diff
@@ -147,7 +147,21 @@ pub struct BeaconState {
 }
 
 impl BeaconState {
-    pub fn from_ssz_bytes(bytes: &[u8], fork_name: ForkName) -> Result<Self, DecodeError> {
+    pub fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
```

Comment:
My only concern about using this pattern is that it is slow because the beacon state is huge. This is why I was trying to avoid implementing it. (this is also mentioned in the lighthouse docs, I don't remember where exactly).

Can we get away by using the explicit version and get information about the era fork before deserializing the beacon state? For example, if we know the slot of the block we can determine which fork it is part of.